internal/bminventory/inventory.go

```diff
@@ -2175,6 +2254,113 @@ func (b *bareMetalInventory) updateClusterInternal(ctx context.Context, params i
 	return cluster, nil
 }
 
+func (b *bareMetalInventory) v2updateClusterInternal(ctx context.Context, params installer.V2UpdateClusterParams, interactivity Interactivity) (*common.Cluster, error) {
+	log := logutil.FromContext(ctx, b.log)
+	var cluster *common.Cluster
+	var err error
+	log.Infof("update cluster %s with params: %+v", params.ClusterID, params.ClusterUpdateParams)
+
+	if params, err = b.validateAndUpdateClusterV2Params(ctx, &params); err != nil {
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	txSuccess := false
+	tx := b.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("update cluster failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("update cluster failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Errorf("failed to start db transaction")
+		return nil, common.NewApiError(http.StatusInternalServerError,
+			errors.New("DB error, failed to start transaction"))
+	}
+
+	// in case host monitor already updated the state we need to use FOR UPDATE option
+	tx = transaction.AddForUpdateQueryOption(tx)
+
+	if cluster, err = common.GetClusterFromDB(tx, params.ClusterID, common.UseEagerLoading); err != nil {
+		log.WithError(err).Errorf("failed to get cluster: %s", params.ClusterID)
+		return nil, common.NewApiError(http.StatusNotFound, err)
+	}
+
+	if params, err = b.validateAndUpdateProxyV2Params(ctx, &params, &cluster.OpenshiftVersion); err != nil {
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	if err = b.clusterApi.VerifyClusterUpdatability(cluster); err != nil {
+		log.WithError(err).Errorf("cluster %s can't be updated in current state", params.ClusterID)
+		return nil, common.NewApiError(http.StatusConflict, err)
+	}
+
+	if err = b.noneHaModeV2ClusterUpdateValidations(cluster, params); err != nil {
+		log.WithError(err).Warnf("Unsupported update params in none ha mode")
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	if err = b.v2ValidateDNSDomain(*cluster, params, log); err != nil {
+		return nil, err
+	}
```

Comment:
what about DiskEncryption validation? Has it been moved to another location?

Reply:
Fixed